presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -156,6 +170,17 @@ private Query getRelationQuery(ShowStats node)
             throw new IllegalArgumentException("Expected either TableSubquery or Table as relation");
         }
 
+        private Table getTable(ShowStats node, QuerySpecification specification)
+        {
+            check(specification.getFrom().isPresent(), node, "There must be exactly one table in query passed to SHOW STATS SELECT clause");
+            List<Table> tables = preOrder(specification.getFrom().get())
+                    .filter(Table.class::isInstance)
```

Comment:
This is fishy. Rather than _filter_, we should _verify_ the  the FROM is just a table.
(like we did https://github.com/prestosql/presto/blob/1f9088dc5757d18abf1f622f64d8bad4a87e8e2b/presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java#L176-L177)

Add a test case like

```
STATS FOR 
  SELECT * 
  FROM nation, (subquery)
```